lightning/src/util/config.rs

```diff
@@ -893,3 +893,157 @@ impl Readable for UserConfig {
 		})
 	}
 }
+
+/// Config structure for overriding incoming channel parameters.
+#[derive(Default)]
+pub struct InboundChannelConfigOverrides {
+	/// Overrides for incoming channel handshake parameters.
+	pub handshake_overrides: Option<InboundChannelHandshakeConfigOverrides>,
+
+	/// Overrides for incoming channel update parameters.
+	pub update_overrides: Option<InboundChannelUpdateOverrides>,
+}
+
+/// Config structure for overriding incoming channel handshake parameters.
+#[derive(Default)]
+pub struct InboundChannelHandshakeConfigOverrides {
+	/// Overrides the percentage of the channel value we will cap the total value of outstanding inbound HTLCs to. See
+	/// [`ChannelHandshakeConfig::max_inbound_htlc_value_in_flight_percent_of_channel`].
+	pub max_inbound_htlc_value_in_flight_percent_of_channel: Option<u8>,
+
+	/// Overrides the smallest value HTLC we will accept to process. See [`ChannelHandshakeConfig::our_htlc_minimum_msat`].
+	pub htlc_minimum_msat: Option<u64>,
+
+	/// Overrides confirmations we will wait for before considering the channel locked in. See
+	/// [`ChannelHandshakeConfig::minimum_depth`].
+	pub minimum_depth: Option<u32>,
+
+	/// Overrides the number of blocks we require our counterparty to wait to claim their money. See
+	/// [`ChannelHandshakeConfig::our_to_self_delay`].
+	pub to_self_delay: Option<u16>,
+
+	/// The maximum number of HTLCs in-flight from our counterparty towards us at the same time. See
+	/// [`ChannelHandshakeConfig::our_max_accepted_htlcs`].
+	pub max_accepted_htlcs: Option<u16>,
+
+	/// The Proportion of the channel value to configure as counterparty's channel reserve. See
+	/// [`ChannelHandshakeConfig::their_channel_reserve_proportional_millionths`].
+	pub channel_reserve_proportional_millionths: Option<u32>,
+}
+
+/// Config structure for overriding incoming channel configuration parameters.
+#[derive(Default)]
+pub struct InboundChannelUpdateOverrides {
```

Comment:
This resembles the pre-existing `ChannelConfigUpdate`, but that struct doesn't have `accept_underpaying_htlcs`. So created a new one here.

Reply:
Good points, fixed.